src/asset.rs

```diff
@@ -0,0 +1,286 @@
+use crate::prelude::*;
+
+#[derive(Clone, Debug)]
+pub struct Asset {
+    /// identifier of asset, formatted as asset_name#domain_id
+    pub id: Id,
+}
+
+impl Asset {
+    pub fn new(id: Id) -> Self {
+        Asset { id }
+    }
+}
+
+pub mod isi {
+    use super::*;
+    use crate::isi::Command;
+
+    /// The purpose of add asset quantity command is to increase the quantity of an asset on account of
+    /// transaction creator. Use case scenario is to increase the number of a mutable asset in the
+    /// system, which can act as a claim on a commodity (e.g. money, gold, etc.).
+    #[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+    pub struct AddAssetQuantity {
+        pub asset_id: Id,
+        pub account_id: Id,
+        //TODO[@humb1t:RH2-11]: decide which format to use in such a case
+        //value can be non-integer, but should be precise.
+        pub amount: f64,
```

Comment:
this shouldn't be floating point

Instead you just set decimal precision as a separate number. Like "decimals"